### ISSUE TITLE
New element vectorfield

### ISSUE THREAD

**Issue by alfredwassermann** [jsxgraph/jsxgraph#333]
We are planning to introduce a new element `vectorfield`.
Please, make suggestion here, how such an element should look like. 
What kind of input parameters are useful?

    board.create('vectorfield`, [????]);

**Comment by davidf628** [comment]
Alfred, in the past I have written apps for both vector fields and for slope fields, both of which are very related and it might be a reasonable attribute to be able to use one or the other. For vector fields in particular, the variations I would find useful would be:
1. The ability to set the length of vector to either a scaled value of length, or ability to make all vectors unit length
2. For unit length vectors, another useful aspect is to use different colors to represent magnitude that would be a nice option
3. Certainly the ability to control the width of the vectors, whether or not they have arrows and how large the arrows are
For inputs, of course two functions P(x, y) and Q(x, y) are required parameters, but in addition it is a good option to allow the user how much x and y distance from one vector to the next there should be. 

That is all I can think of, I hope it is helpful.

Best,
David Flenner

**Comment by derivitaandrew** [comment]
It would make sense for `vectorfield` to be constructed by `board.create('vectorfield', [f])`, where `f` is a function of x and y.

Users should certainly be able to define how frequently to draw the vectors. It seems more useful to choose the step size between samples, rather than choosing how many samples total, though the latter might be useful if you want to zoom in and out on the vector field without the arrows becoming to sparse or too crowded.

I could imagine it having optional elements for the parent array that define the range of x- and y-values to use--the rectangle over which to sample--but I wouldn't personally use that, since whenever I'm making a vector field, I want it to take up the whole window.

**Comment by soegaard** [comment]
David's list covers most uses.

In the nice-to-have department an option that effects which points are sampled would be nice.
I notice that Mathematica samples points in a triangular grid:

<img width="764" alt="image" src="https://user-images.githubusercontent.com/461765/210136908-977cf34b-a6e3-4c23-91cf-ba761cee0f40.png">

I found the image under "Details and options" here:
https://reference.wolfram.com/language/ref/VectorPlot.html

Apropos Mathematica, I really like the look of their stream plots:
https://reference.wolfram.com/language/ref/StreamPlot.html

**Comment by axkr** [comment]
The Mathematica `Plot` functions internally generate `Graphics` output.
- https://reference.wolfram.com/language/ref/Graphics.html

The `VectorPlot` and `StreamPlot` especially generates `Arrowheads` "graphic primitives":
- https://reference.wolfram.com/language/ref/Arrowheads.html

Currently there is a "work-in-progress" for implementning "JSON 2D Graphics interface for JSXGraph" for these primitives (`Arrowheads` isn't included yet):
- https://github.com/jsxgraph/json2D_JSXGraph

Currently the WIP can be tested in the demo for some plot functions like `Plot,PolarPlot, ListPlot, ListLogPlot,...` (though some "key features like `Log` scales are missing):
- http://matheclipse.org/

**Comment by soegaard** [comment]
FWIW when it comes to implementing arrow heads, I can recommend 
Hans van der Meer's paper: "Blocs and Arrows with MetaPost". [1]

I followed the paper, when I implemented arrow heads  for Metapict [2].
Curves in Metapict are specified using the same idea as the curve notation in MetaPost.
And JSXGraph contains an implementation of the Hobby's algorithm.

[1]  https://staff.fnwi.uva.nl/h.vandermeer/pubs/blockarrowmaps.pdf
[2] https://youtu.be/7Kf8Kk8NPFw?t=318
